transmogrifier/sources/json/aardvark.py

```diff
@@ -368,11 +369,14 @@ def get_publishers(source_record: dict) -> list[timdex.Publisher]:
                     for publisher in source_record["dct_publisher_sm"]
                 ]
             )
-        return publishers
+        return publishers or None
 
-    @staticmethod
-    def get_rights(source: str, source_record: dict) -> list[timdex.Rights]:
-        """Get values from source record for TIMDEX rights field."""
+    @classmethod
+    def get_provider(cls, source_record: dict) -> str | None:
+        return source_record.get("schema_provider_s") or None
+
+    @classmethod
+    def get_rights(cls, source_record: dict, source: str) -> list[timdex.Rights] | None:
```

Comment:
Perhaps this should be broken into submethods but I wanted to get your thoughts to see if it's worth it

Reply:
I hate to say it, but I think I'd agree.  I could see methods like:
- `get_access_rights()`
- `get_license_rights()`
- `get_rights_and_rights_holders()`
- `get_gismit_rights()` + `get_gisogm_rights()` (and perhaps these triggered by what source logic)

An intersting one for sure though, it's like a big collapsing of multiple fields into one.

Reply:
Deep down, I knew it had to be done 🙃